Mercury-Store/Mercury-Store/Profile/Views/LoginView/LoginViewController.swift

```diff
@@ -22,35 +30,32 @@ class LoginViewController: UIViewController,ProfileCoordinated {
     }
     
    
-    @IBOutlet weak var emailTextField: UITextField!
-    
-    @IBOutlet weak var signupLabel: UILabel!
-    @IBOutlet weak var passwordTextField: UITextField!
+   
     override func viewDidLoad() {
         super.viewDidLoad()
+        emailTextField.becomeFirstResponder()
+        emailTextField.rx.text.map{ $0 ?? ""}.bind(to:loginViewModel.emailTextPublishSubject).disposed(by: disposeBag)
+        passwordTextField.rx.text.map{ $0 ?? ""}.bind(to:loginViewModel.passwordTextPublishSubject).disposed(by: disposeBag)
+        loginViewModel.isValid().bind(to: loginButton.rx.isEnabled).disposed(by: disposeBag)
+        loginViewModel.isValid().map{$0 ? 1: 1.0}.bind(to: loginButton.rx.alpha).disposed(by: disposeBag)
 
```

Comment:
Separate methods and each one should handle common logic

```suggestion
        emailTextField.becomeFirstResponder()
        bindTextToViewModel()
        observeViewModelOnValid()
    }
    
    func bindTextToViewModel() {
        emailTextField
            .rx.text
            .map { $0 ?? "" }
            .bind(to: loginViewModel.emailTextPublishSubject)
            .disposed(by: disposeBag)
        
        passwordTextField
            .rx.text
            .map { $0 ?? "" }
            .bind(to: loginViewModel.passwordTextPublishSubject)
            .disposed(by: disposeBag)
    }
    
    func observeViewModelOnValid() {
        loginViewModel
            .isValid()
            .bind(to: loginButton.rx.isEnabled)
            .disposed(by: disposeBag)
        
        loginViewModel
            .isValid()
            .map{ $0 ? 1 : 1.0 }
            .bind(to: loginButton.rx.alpha)
            .disposed(by: disposeBag)
    }
```

Reply:
resolved